src/lib/__tests__/insights-client-test.ts

```diff
@@ -0,0 +1,231 @@
+import { withInsights, inferInsightsPayload } from '../insights';
+
+const connectHits = (renderFn: any, unmountFn: any) => (widgetParams = {}) => ({
+  init() {},
+  render({ results, instantSearchInstance }) {
+    const hits = results.hits;
+    renderFn({ hits, results, instantSearchInstance, widgetParams }, false);
+  },
+  dispose() {
+    unmountFn();
+  },
+});
+
+describe('withInsights', () => {
+  let instantSearchInstance: any;
+  let hits: any;
+  let results: any;
+  let widgetParams: any;
+  let widget: any;
+  let renderFn: any;
+
+  beforeEach(() => {
```

Comment:
We shouldn't set so many things in the `beforeEach` hook. Tests become hard to follow.

Reply:
the alternative is to move what's inside the beforeEach inside the tests themselves.
It's redundant and makes the tests bigger. IMHO it's not helping the readability.

Reply:
It does IMO and @samouss agreed in another PR comment.

You should be able to read a test just like you read a book. Seeing [this example](https://github.com/algolia/instantsearch.js/blob/7b0595e0a75a7efb29ee2657215f3cefc982b262/src/lib/__tests__/insights-client-test.ts#L70), I have strictly no idea what the context of the test is without going back to `beforeEach`. Besides, it makes the PR hard to review.

These discussions shouldn't even happen in the first place if we agree beforehand on what and how to test our libraries. That's why we need guidelines about this.

Reply:
ok let's talk about this Wednesday. I'll setup a meeting together with @samouss before the standup, feel free to reschedule.